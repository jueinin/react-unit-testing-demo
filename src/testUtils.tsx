import {render, RenderOptions, RenderResult} from '@testing-library/react'
import React from "react";
import {ReactQueryConfigProvider} from "react-query";
import {BrowserRouter} from "react-router-dom";

export const loadingClass = "MuiCircularProgress-root";
const customRender = (ui: React.ReactElement, options?: RenderOptions):RenderResult => {
    return render(ui, {
        wrapper: props => <BrowserRouter>
            <ReactQueryConfigProvider config={{}}>
                {props.children}
                <div id="modalid"></div>
            </ReactQueryConfigProvider>
        </BrowserRouter>,
        ...options
    })
};
/**
 * @description 单文件内mock模块，本文件生效，需要在describe之前调用本函数
 * @param moduleName
 * @param partial
 */
export const partialMockInFile = (moduleName:string,partial: any) => {
    return jest.mock(moduleName,()=>{
        const actual = jest.requireActual(moduleName)
        return {
            __esModule: true,
            ...actual,
            ...partial
        }
    })
}
export const partialMockInTestCase = (moduleName:string,partial: any) => {
    return jest.doMock(moduleName,()=>{
        const actual = jest.requireActual(moduleName)
        return {
            __esModule: true,
            ...actual,
            ...partial
        }
    })
}
export * from '@testing-library/react'
export {customRender as render}
