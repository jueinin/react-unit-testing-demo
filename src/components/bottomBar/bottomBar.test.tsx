import {render} from "../../testUtils";
import BottomBar from "./bottomBar";
import React from "react";
import {fireEvent} from "@testing-library/dom";
describe("bottom bar",()=> {
    it('should ', function () {
        const wrapper = render(<BottomBar currentValue={"/me"}/>)
        expect(wrapper.getByText(/我的/)).toHaveClass("Mui-selected")
        expect(wrapper.getByText(/论坛/)).not.toHaveClass("Mui-selected")
        expect(wrapper.getByText(/首页/)).not.toHaveClass("Mui-selected")
        fireEvent.click(wrapper.getByText(/论坛/)) //点击 跳转路由
        expect(window.location.pathname).toEqual("/forum")
    });
})