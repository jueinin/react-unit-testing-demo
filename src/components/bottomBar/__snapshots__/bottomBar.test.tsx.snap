// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottom bar should  1`] = `
<div>
  <div
    class="MuiBottomNavigation-root mt-auto fixed bottom-0 w-full"
  >
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          首页
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          论坛
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label Mui-selected"
        >
          我的
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
